src/core/packages/chrome/layout/core-chrome-layout/layouts/grid/grid_global_app_style.tsx

```diff
@@ -85,42 +88,33 @@ const globalTempHackStyles = (euiTheme: UseEuiTheme['euiTheme']) => css`
     ${logicalCSS('bottom', 'var(--kbn-layout--application-bottom, 0px)')};
   }
 
-  // adjust position of all other flyouts
-  .kbnBody .euiFlyout:not(.euiCollapsibleNav) {
-    // overlay flyout should only cover the application area
-    &[class*='right']:not([class*='push']) {
+  .kbnBody {
+    // adjust position of all the right flyouts relative to the application area, except the ones that are "above the header"
+    .euiFlyout[class*='right']:not(.euiOverlayMask[class*='aboveHeader'] .euiFlyout) {
       ${logicalCSS('top', 'var(--kbn-layout--application-top, 0px)')};
       ${logicalCSS('bottom', 'var(--kbn-layout--application-bottom, 0px)')};
       ${logicalCSS('right', 'var(--kbn-layout--application-right, 0px)')};
     }
-    // push flyout should only cover the application area
-    &[class*='right'][class*='push'] {
+
+    // overlay mask "belowHeader" should only cover the application area
+    .euiOverlayMask[class*='belowHeader'] {
       ${logicalCSS('top', 'var(--kbn-layout--application-top, 0px)')};
-      ${logicalCSS('bottom', 'var(--kbn-layout--application-bottom, 0px)')};
+      ${logicalCSS('left', 'var(--kbn-layout--application-left, 0px)')};
       ${logicalCSS('right', 'var(--kbn-layout--application-right, 0px)')};
+      ${logicalCSS('bottom', 'var(--kbn-layout--application-bottom, 0px)')};
     }
-
-    // ...
-    // no use-cases for left flyouts other then .euiCollapsibleNav, so skipping it for now
   }
 
   // push flyout should be pushing the application area, instead of body
-  main[class*='LayoutApplication'] {
+  main#${APP_MAIN_SCROLL_CONTAINER_ID} {
```

Comment:
This is an important fix for the "build" of Kibana. It turns out the emotion part `LayoutApplication` gets removed in the production build, and we can't use it as a selector.